service/src/java/org/apache/hive/service/cli/operation/OperationManager.java

```diff
@@ -85,10 +79,7 @@ public synchronized void init(HiveConf hiveConf) {
     LogDivertAppender.registerRoutingAppender(hiveConf);
     LogDivertAppenderForTest.registerRoutingAppenderIfInTest(hiveConf);
 
-    if (hiveConf.isWebUiQueryInfoCacheEnabled()) {
-      historicalQueryInfos = new QueryInfoCache(
-        hiveConf.getIntVar(ConfVars.HIVE_SERVER2_WEBUI_MAX_HISTORIC_QUERIES));
-    }
+    this.queryInfoCache = new QueryInfoCache(hiveConf);
```

Comment:
Do we need `QueryInfoCache` when we does not use the WebUI?
Would it be possible to use `Optional` for it?

Reply:
yes, the cache will only be used when the WebUI is enabled, make it Optional here.